core/src/com/unciv/logic/civilization/CivInfoStats.kt

```diff
@@ -200,9 +200,9 @@ class CivInfoStats(val civInfo: CivilizationInfo) {
             statMap["Religion"] = 0f
             for (unique in civInfo.religionManager.religion!!.getFounderBeliefs().flatMap { it.uniqueObjects }) {
                 if (unique.placeholderText == "[] for each global city following this religion") {
-                    statMap["Religion"] = statMap["Religion"]!! + unique.stats.apply {
-                        this.timesInPlace(civInfo.religionManager.numberOfCitiesFollowingThisReligion().toFloat())
-                    }.happiness
+                    statMap["Religion"] = 
+                        statMap["Religion"]!! +
```

Comment:
:+1: .  Really makes one think, though. There's a reason why so many CPU architectures have MulAdd opcodes. Plus I still think our Stats with its permanent toHashMap (meaning tons of small allocations and work for the GC) is - not nice. And - why does StatMap.add _not_ pull a clone when the addition is new, but _does_ pull a clone when it's already in? I'd have opted for the other way round.

Reply:
I'm pretty sure that stats.add goes over all values and adds them as ints for efficiency?
Also, statMap.add("Religion", stats) exists for this reason...

Reply:
Except that this `statMap` isn't a `statMap` but instead an `hashMap`, so it doesn't have that function sadly. Why it's still called `statMap` I don't know